### PR TITLE
[#720] Update beta kubernetes objects to stable versions

### LINKED ISSUE #720: Update beta kubernetes objects to stable versions
We should upgrade all beta kubernetes object to stable version.
For example:
* model `ExtensionsV1beta1Deployment` to `V1Deployment`
* edi\edge `ReplicationController` to `V1Deployment`
* Role and RoleBinding `rbac.authorization.k8s.io/v1beta1` to `rbac.authorization.k8s.io/v1`

and so on.

### PR DESCRIPTION
**Pull request by skayerok**
This closes #720

### CODE REVIEW

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

File: deploy/helms/legion/templates/edi/edi-rbac.yaml [hunk 1]
```diff
@@ -16,10 +16,10 @@ rules:
   verbs:  ["list", "create", "patch", "get"]
 - apiGroups: ["apps"]
   resources: ["deployments"]
-  verbs:  ["list", "delete", "patch", "get"]
+  verbs:  ["list", "delete", "patch", "get", "create"]
```
Comment (kirillmakhonin) [inline deploy/helms/legion/templates/edi/edi-rbac.yaml:19]: Why did you add new action?

Comment (botsman) [reply deploy/helms/legion/templates/edi/edi-rbac.yaml:19]: There are a lot of tests failing without this action: https://cc.epm.kharlamov.biz/jenkins/job/Deploy_Legion_Release/369/robot/report/log.html#s1-s1-t22-k2-k3
Not quite sure why this happened after version change, but this action solves the problem.

Comment (botsman) [reply deploy/helms/legion/templates/edi/edi-rbac.yaml:19]: Seems that now I understand why this helped.
Edi before was configured as `ReplicationController`, and now it is configured as `Deployment`.

That is why now another rule is applied. Before there were rules from `extensions` `apiGroups`, and not it is rules from `apps`.

Comment (kirillmakhonin) [reply deploy/helms/legion/templates/edi/edi-rbac.yaml:19]: It this is true, we have to remove RBAC rules for `ReplicationController`s, right? And update docs.

**Comment by skayerok** [comment]
bot, test all

File: deploy/helms/legion/templates/edi/edi-rbac.yaml (hunk 1 above)
Comment (botsman) [reply deploy/helms/legion/templates/edi/edi-rbac.yaml:19]: If I understand everything correctly, we have to remove `deployments` from `apiGroups: ["extensions"]` (just did it).
Since Ingress is still configured through `extensions/v1beta1`, the rest of rule for `extensions` should remain the same.

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by botsman** [comment]
bot, test all
